hi-ml-histopathology/src/histopathology/utils/metrics_utils.py

```diff
@@ -155,8 +155,14 @@ def plot_heatmap_overlay(slide: str,
 
     # for each tile in the bag
     for tile_idx in range(len(results[ResultsKey.IMAGE_PATH][slide_idx])):
-        tile_coords = np.transpose(np.array([results[ResultsKey.TILE_X][slide_idx][tile_idx].cpu().numpy(),
-                                             results[ResultsKey.TILE_Y][slide_idx][tile_idx].cpu().numpy()]))
+        if (ResultsKey.TILE_TOP in results.keys()) and (ResultsKey.TILE_LEFT in results.keys()) \
+                and (ResultsKey.TILE_RIGHT in results.keys()) and (ResultsKey.TILE_BOTTOM in results.keys()):
+            tile_coords = np.transpose(np.array([results[ResultsKey.TILE_LEFT][slide_idx][tile_idx].cpu().numpy(),
```

Comment:
[Optional] I think calling `np.stack([x, y], axis=1)` would make the intent and output shape clearer than `np.transpose(np.array([x, y]))` (also below and in `test_metrics_utils.py`).

Reply:
Optional: There's quite a lot repetition and nested indexing, which makes this code a bit hard to read and understand. For example, x and y could be defined in the if/else block and stacking could happen after.

Reply:
Thanks, the nesting (condition) is now removed as we don't have `ResultsKey.TILE_X/Y` now, as suggested